### PR TITLE
GYE 1943

### PR DESCRIPTION
**Pull request by Jodlowy**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/emailfrontend/connectors/SubscriptionDisplayConnector.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@Singleton
+class SubscriptionDisplayConnector @Inject()(appConfig: AppConfig, http: HttpClient, auditable: Auditable) {
+
+  private[connectors] lazy val url: String = appConfig.subscriptionDisplayUrl
+
+  def subscriptionDisplay(eori: Eori)(implicit hc: HeaderCarrier): Future[SubscriptionDisplayResponse] = {
+    val request = ("EORI" -> eori.id) :: buildQueryParams
+
+    http.GET[SubscriptionDisplayResponse](url, request).map { displayResponse =>
+      auditResponse("customs-email-subscription-display", "subscriptionDisplayRequest", displayResponse.email, url)
```
Comment (kapil-sanghvi) [inline app/uk/gov/hmrc/customs/emailfrontend/connectors/SubscriptionDisplayConnector.scala]: auditType should be `subscriptionDisplayResponse`

File: app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala [hunk 2]
```diff
@@ -47,17 +51,21 @@ class WhatIsYourEmailController @Inject()(actions: Actions, view: what_is_your_e
...
+      case SubscriptionDisplayResponse(Some(email)) => Ok(view(emailForm, email))
+    }
   }
 
-  def submit: Action[AnyContent] = actions.auth.async { implicit request =>
+  def submit: Action[AnyContent] = (actions.auth andThen actions.eori).async { implicit request =>
     emailForm.bindFromRequest.fold(
       formWithErrors => {
-        Future.successful(BadRequest(view(formWithErrors)))
+        subscriptionDisplayConnector.subscriptionDisplay(Eori(request.eori.id)).map {
```
Comment (kapil-sanghvi) [inline app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala]: do we need to call subscription display (ETMP) again while submitting the form and just to show the errors on form page, doesn't feel great about this @shaileshp0110

File: app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala [hunk 3]
```diff
@@ -47,17 +51,21 @@ class WhatIsYourEmailController @Inject()(actions: Actions, view: what_is_your_e
...
+          case SubscriptionDisplayResponse(Some(email)) => BadRequest(view(formWithErrors, email))
```
Comment (kapil-sanghvi) [inline app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala]: not exhaustively matched

File: build.sbt [hunk 4]
```diff
@@ -52,7 +52,7 @@ lazy val scoverageSettings = {
     "uk\\.gov\\.hmrc\\.customs\\.emailfrontend\\.views\\.html\\.partials*",
     "uk\\.gov\\.hmrc\\.customs\\.emailfrontend\\.views\\.html\\.helpers*",
     ".*(BuildInfo|Routes|TestOnly).*").mkString(";"),
-    ScoverageKeys.coverageMinimum := 98.50,
+    ScoverageKeys.coverageMinimum := 98.8,
```
Comment (kapil-sanghvi) [inline build.sbt]: great stuff

File: test/acceptance/specs/EmailConfirmedSpec.scala [hunk 5]
```diff
@@ -66,7 +68,8 @@ class EmailConfirmedSpec extends BaseSpec
       save4LaterWithNoData("123456789")
       navigateTo(StartPage)
       verifyCurrentPage(StartPage)
-      authenticateGGUserAndReturnEoriEnrolment("123456789")
+      stubSubscriptionDisplayOkResponse()
```
Comment (kapil-sanghvi) [inline test/acceptance/specs/EmailConfirmedSpec.scala]: can we verify that the number of calls made to subscription display(customs-hods-proxy) here, please

File: test/acceptance/specs/SendEmailVerificationSpec.scala [hunk 6]
```diff
@@ -33,7 +34,8 @@ class SendEmailVerificationSpec extends BaseSpec
         save4LaterWithNoData("111111111")
         navigateTo(StartPage)
         verifyCurrentPage(StartPage)
-        authenticateGG("111111111")
+        authenticateGGUserAndReturnEoriEnrolment("111111111", "111111111")
```
Comment (kapil-sanghvi) [inline test/acceptance/specs/SendEmailVerificationSpec.scala]: can we pass different value for `internalId` for differntiation purpose with eoriNumber value

File: test/acceptance/utils/StubSubscriptionDisplay.scala [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        "contactInformation": {
+        "emailAddress": "b@a.com",
+        "emailVerificationTimestamp": "2019-09-06T12:30:59Z"
+            }
+          }
+        }
+    }""".stripMargin
+
+  def stubSubscriptionDisplayOkResponse(): Unit = {
+    stubFor(get(urlPathEqualTo(s"$subscriptionDisplay"))
```
Comment (kapil-sanghvi) [inline test/acceptance/utils/StubSubscriptionDisplay.scala]: we should be able to verify url query params here, maybe (Ref: `SubscriptionDisplayMessagingService` in customs-rosm-frontend)

File: app/uk/gov/hmrc/customs/emailfrontend/connectors/SubscriptionDisplayConnector.scala (hunk 1 above)
Comment (Jodlowy) [reply app/uk/gov/hmrc/customs/emailfrontend/connectors/SubscriptionDisplayConnector.scala]: done

File: app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala (hunk 3 above)
Comment (Jodlowy) [reply app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala]: yes I know, but the story is  for a happy path only

File: build.sbt (hunk 4 above)
Comment (Jodlowy) [reply build.sbt]: Indeed!

File: test/acceptance/specs/EmailConfirmedSpec.scala (hunk 5 above)
Comment (RashmithReddy) [reply test/acceptance/specs/EmailConfirmedSpec.scala]: will be addressed as part of testing

File: test/acceptance/specs/SendEmailVerificationSpec.scala (hunk 6 above)
Comment (RashmithReddy) [reply test/acceptance/specs/SendEmailVerificationSpec.scala]: will be addressed as part of testing

File: test/acceptance/utils/StubSubscriptionDisplay.scala (hunk 7 above)
Comment (RashmithReddy) [reply test/acceptance/utils/StubSubscriptionDisplay.scala]: will be addressed as part of testing

**Pull request merged**